### PR TITLE
SgV2: Separate "rows" endpoint into API interface, impl class

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Separates formerly coupled REST endpoint API and implementation into interface (API, annotations) and implementation (no method annotations, actual implementation)

**Checklist**
- [x] Changes manually tested -- verify Swagger API works as well (no unit tests)
- [x] Automated Tests added/updated -- passes same tests as before changes (all endpoints covered)
- [ ] Documentation added/updated

### CODE REVIEW

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResourceApi.java [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+      value = {
+        @ApiResponse(code = 200, message = "OK", response = Sgv2RowsResponse.class),
+        @ApiResponse(code = 400, message = "Bad Request", response = RestServiceError.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = RestServiceError.class),
+        @ApiResponse(
+            code = 500,
+            message = "Internal server error",
+            response = RestServiceError.class)
+      })
+  public Response getRowWithWhere(
```
Comment (olim7t) [inline sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResourceApi.java:52]: Nit: `public` can be omitted in interface definitions, it's implied.

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResourceApi.java [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+          String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(value = "Name of the table to use for the request.", required = true)
+          @PathParam("tableName")
+          final String tableName,
+      @ApiParam(
+              value =
+                  "Value from the primary key column for the table. Define composite keys by separating values with slashes (`val1/val2...`) in the order they were defined. </br> For example, if the composite key was defined as `PRIMARY KEY(race_year, race_name)` then the primary key in the path would be `race_year/race_name` ",
```
Comment (olim7t) [inline sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResourceApi.java:128]: Can we split this with line breaks?

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResourceApi.java [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.PathSegment;
+import javax.ws.rs.core.Response;
+
+/**
+ * Definition of REST API endpoint methods including both JAX-RS and Swagger annotations. No
+ * implementations.
+ *
+ * <p>NOTE: class annotations cannot be included in the interface (at least for JAX-RS) and must be
+ * included in the implementation class.
```
Comment (olim7t) [inline sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResourceApi.java:32]: `@Api` can be inherited, so maybe we can move it to the interface in order to have all the Swagger annotations in one place.

**Review by olim7t — APPROVED** [review]
👍 
I guess a point could be made that the Swagger annotations also document the code, but they're really too verbose and end up obscuring rather than helping.
So I prefer this approach to make the implementation easier to follow. If the parameters have good names, this should be enough to document their purpose.

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResourceApi.java (hunk 2 above)
Comment (tatu-at-datastax) [reply sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResourceApi.java:128]: I can try to see how Google indenter handles that.

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResourceApi.java (hunk 3 above)
Comment (tatu-at-datastax) [reply sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResourceApi.java:32]: Class annotation `@Api`? Ok, I can move that, makes sense. I only checked that JAX-RS ones cannot be moved.

**Comment by tatu-at-datastax** [comment]
Agreed, it's not an optimal solution. But when just annotations produce 1/3 of all lines (partly due to auto-indenter using... lots of space) it does seem better to split things as that makes it easier to change implementation, where most work usually is.

I made suggested changes and will merge this; we can further improve aspects as things go.

Couple of things to note:

1. Only `interface` works: I tried abstract class and that did NOT work with Jersey at least (although some docs suggested it might)
2. While JAX-RS class annotations must be on impl class, Swagger seems smarter and allows annotations on either (but like JAX-RS, probably won't merge if both have them).

**Pull request merged**